Ch 73 stylize the search page

timDeHof opened the pull request:
Stylized the search page to make it similar to the figma design

timDeHof committed:
chore(package.json): remove unused tailwindcss-vite-plugin dependency

The tailwindcss-vite-plugin was removed as it was not being used, reducing the project's
dependencies.

CH-73

timDeHof committed:
style(searchbox.jsx): redesign search box for bette user experience

The CSS classes in SearchSection.jsx were refactored for better readability and consistency.
Unnecessary console.log statements were removed from recipe-list.jsx to clean up the code. The
search box in SearchBox.jsx was redesigned to improve the user experience.

CH-73

timDeHof committed:
feat(search.jsx): add drag prevention for Yumi character and redesign page layout

Drag prevention was added to the Yumi character in Search.jsx to prevent unwanted user interactions,
and the page layout was redesigned for better aesthetics. A new text-shadow class was added to
global.css for future use, and a new scale value was added to tailwind.config.js for future use.

Ch-73

timDeHof committed:
Merge branch 'development' into CH-73-stylize-the-search-page

timDeHof committed:
Merge branch 'development' into CH-73-stylize-the-search-page

timDeHof committed:
feat(jsconfig.json): add path alias for assets directory

The addition of a new path alias for the assets directory in jsconfig.json allows for easier and
more readable imports of assets in the codebase. This change enhances code readability and
maintainability.

CH-73

timDeHof committed:
feat(tailwind.config.js): add gradient-Search to tailwind configuration

A new gradient named "gradient-Search" has been added to the tailwind configuration. This new
gradient will provide more styling options for the UI, particularly for search-related components,
enhancing the overall user experience.

timDeHof committed:
feat(assets): add new FeatureRecipe image and export it in index.js

A new image file named 'Recipe Photo.png' has been added to the assets directory. This image is
intended to be used as a featured recipe image on the website. The image has been imported and
exported in the assets index.js file to make it available for use throughout the application.

timDeHof committed:
feat(assets): add new Sweet Peppers brand assets in PNG, PSD, and WEBP formats

The new Sweet Peppers brand assets have been added in PNG, PSD, and WEBP formats. This provides a
variety of image formats for different use cases, ensuring optimal compatibility and performance
across different platforms and devices. The PNG and WEBP formats include multiple resolutions to
support responsive design. The PSD file is included for any necessary modifications or adaptations
in the future.

timDeHof committed:
fix(svgwave.jsx): add fallback color to SVG fill property

The SVG fill color now has a fallback to the currentColor property. This change ensures that the SVG
will still have a color even if the primary color (#FFC945) fails to load, enhancing the reliability
of the SVG rendering.

timDeHof committed:
refactor(feature-of-the-day.jsx): add image to the feature of the day section and improved layout

The commented code was removed to improve code readability and maintainability. The layout was also
adjusted to improve the user experience. An image was added to the feature of the day section to
make it more visually appealing and informative. The button was also updated to improve its
appearance and responsiveness.

timDeHof committed:
refactor(recipe-list.jsx): update CSS classes for better layout and responsiveness

The CSS classes in the recipe-list component have been updated to improve the layout and make it
more responsive. The changes include padding adjustments and grid column configurations for
different screen sizes. Additionally, console logs have been added to help trace the recipe data and
debug any issues related to it.

timDeHof committed:
style(recipe-card.jsx): enhance RecipeCard component styling for better UX

The RecipeCard component styling has been updated to improve user experience. The card now has a
hover effect that scales the card to 110% of its original size, providing a more interactive feel.
The image height has been increased to 200px for better visibility. The text alignment has been
changed to center for better readability and consistency. The gap between elements in the flex
container has been reduced to 4px to make the layout more compact. The font size of the servings
text has been increased for better readability. The tracking (letter spacing) of the heading has
been tightened for a cleaner look. The padding on the x-axis of the flex container holding the
servings text and star rating has been increased to 8px to provide more breathing space.

timDeHof committed:
style(searchbox.jsx): adjust input field and button styles for better responsiveness and usability

The input field now takes up the full width of its container, and its padding and font size have
been adjusted for better readability and usability on different screen sizes. The search button has
been updated to use inline-flex for better alignment with the input field, and its width has been
fixed to ensure consistent appearance across different screen sizes. The search icon's width has
also been set to maintain its proportions when scaling. These changes improve the overall user
experience and make the search box more responsive and usable on different devices.

timDeHof committed:
feat(search.jsx): redesign search page layout and add new SVG component

The search page layout has been redesigned to improve user experience and aesthetics. A new SVG
component has been added to enhance the visual appeal of the page. The gradient for the search page
has been updated in the global styles to match the new design.

timDeHof committed:
Merge branch 'development' into CH-73-stylize-the-search-page

timDeHof committed:
Merge branch 'development' into CH-73-stylize-the-search-page

timDeHof committed:
Merge branch 'development' into CH-73-stylize-the-search-page

timDeHof committed:
feat(assets/index.js): add FeatureRecipe to exported assets

A new asset, FeatureRecipe, has been added to the list of exported assets in assets/index.js. This
change was made to make the FeatureRecipe image available for use across the application.

timDeHof committed:
refactor(search.jsx): replace Yumi and BellPeppers images with SVG components for better scalability

The Yumi and BellPeppers images have been replaced with SVG components, which allows for better
scalability and performance. The layout and styling of the Search page have been adjusted to improve
responsiveness across different screen sizes and enhance the overall visual appeal of the page.

joekotvas commented:
On the deploy preview here, it looks like the front page hero section has lost its yellow wave.

joekotvas commented on src/assets/Recipe Photo.png:
Probably best to use a regular image with sharp corners, then round them with CSS.

joekotvas commented on src/assets/brand/sweet-peppers/png/index.js:
The sweet-peppers folder has been made obsolete by the code inside bell-peppers.

joekotvas commented on src/assets/brand/sweet-peppers/psd/index.js:
The sweet-peppers folder has been made obsolete by the code inside bell-peppers. Also the PSD was removed.

joekotvas commented on src/assets/brand/sweet-peppers/webp/index.js:
The sweet-peppers folder has been made obsolete by the code inside bell-peppers.

joekotvas commented on src/assets/index.js:
Best to avoid spaces in file names.

joekotvas commented on src/assets/HomePage/svgWave.jsx:
I don't think the yellow default value has a function here; there is always a current color inherited from the containing page. Also, as of right now on the front page the currentColor is transparent; we'll need to update that.

timDeHof replied:
fixed: now add className 'fill-{color}' to change wave color

joekotvas commented on src/features/recipes/components/feature-of-the-day/feature-of-the-day.jsx:
We probably need more top padding on this.

joekotvas commented on src/pages/Search.jsx:
```diff
@@ -10,29 +14,47 @@ const Search = () => {
-          <div className="w-full mx-auto">
-            <SearchBox searchTerm={searchTerm} onSearch={handleSearch} />
+      <SvgComponent className="fill-sky-300" />
+      <div className="flex flex-col flex-shrink-0 w-full lg:relative bg-gradient-Search">
```
We need horizontal padding here.

joekotvas reviewed (changes requested):
Great progress! Here are a few notes.

timDeHof committed:
fix(assets/index.js): orrect import path for FeatureRecipe and removed unused sweet-peppers assets

A new RecipePhoto.png has been added to the assets to be used in the application. The unused
sweet-peppers assets have been removed to clean up the codebase and reduce the application size. The
import path for FeatureRecipe in assets/index.js has been corrected to match the new file location.

timDeHof committed:
refactor(svgwave.jsx): remove redundant fill attribute from path element

The fill attribute was removed from the path element because it was not needed. The color of the SVG
is now controlled by the CSS currentColor property, which allows for better flexibility and
consistency across different uses of the SVG.

timDeHof committed:
refactor(herosection.jsx, searchsection.jsx): optimize class order and add form functionality

The class order in heroSection.jsx and SearchSection.jsx has been optimized for better readability
and maintainability. This change makes it easier to understand the styling applied to each element.

timDeHof committed:
style(feature-of-the-day.jsx): add padding-top to the main div to improve layout

The main div in the FeatureOfTheDay component now has a padding-top of 10 units. This change
improves the layout by providing more space at the top, enhancing the visual appeal and readability
of the component.

timDeHof committed:
style(search.jsx): rearrange CSS classes and add padding to improve layout

The CSS classes in Search.jsx were rearranged for better readability and maintainability. Padding
was added to the div element to improve the layout and provide better spacing, enhancing the overall
user experience.

joekotvas reviewed (approved):
Looks great!
